File: frontend/src/types/log.ts
export enum LogLevel {
  DEBUG = 'debug',
  INFO = 'info',
  WARNING = 'warning',
  ERROR = 'error',
  CRITICAL = 'critical'
}

export interface LogEntry {
  id: string;
  session_id: string;
  request_id: string;
  log_level: LogLevel;
  message: string;
  creator_id: string | null;
  agent_id: string;
  created_at: string;
  updated_at: string;
}

export type Log = LogEntry; 

File: backend/src/schemas/ws/log.py
from datetime import datetime
from typing import Optional, Union
from uuid import UUID
from pydantic import BaseModel


class LogBase(BaseModel):
    session_id: Union[str, UUID]
    request_id: Union[str, UUID]
    log_level: str
    message: str


class LogCreate(LogBase):
    creator_id: Optional[str] = None
    agent_id: str


class LogUpdate(LogBase):
    pass


class LogEntry(LogCreate):
    created_at: datetime
    updated_at: datetime


class LogEntryDTO(LogEntry):
    pass


class FrontendLogEntryDTO(BaseModel):
    type: str  # TODO: enum
    log: LogEntry
